src/mscorlib/src/System/Number.cs

```diff
@@ -1211,5 +1277,154 @@ internal unsafe struct NumberBuffer {
 
             return true;
         }
+
+        private unsafe static NumberTryParseResult TryParseUnsignedInvariantInteger(string str, ulong minValue, ulong maxValue, out ulong result)
+        {
+            result = 0;
+            if (str == null)
+            {
+                return NumberTryParseResult.NullString;
+            }
+
+            fixed (char* strPointer = str)
+            {
+                // Consume any leading or trailing whitespace
+                int start = 0;
+                int end = str.Length;
+                while (start < end && IsWhite(strPointer[start]))
+                {
+                    start++;
+                }
+                while (end > start && IsWhite(strPointer[end - 1]))
+                {
+                    end--;
+                }
+
+                if (start == end)
+                {
+                    // Nothing to parse
+                    return NumberTryParseResult.InvalidFormat;
+                }
+
+                for (int i = start; i < end; i++)
+                {
+                    int digitValue = 0;
+                    char c = strPointer[i];
+                    if (c >= '0' && c <= '9')
+                    {
+                        digitValue = c - '0';
+                    }
+                    else
+                    {
+                        // Non-digit char
+                        return NumberTryParseResult.InvalidFormat;
+                    }
+
+                    ulong previousResult = result;
+                    result = (result * 10) + (ulong)digitValue;
```

Comment:
Updating `result` while parsing is likely a bug. If parsing fails `result` will end up containing a non-zero value and that breaks `TryParse`.